lib/argo_cd/client.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+require_relative 'response'
+
+module ArgoCd
+  # Client class for interacting with ArgoAPI
+  class Client
+    attr_accessor :app_id, :deployment_name, :current_user_id
+
+    def initialize(app_id, deployment_name, current_user_id)
+      @app_id = app_id
+      @deployment_name = deployment_name
+      @current_user_id = current_user_id
+    end
+
+    def app_info
+      uri = URI("#{base_path}/api/v1/applications?name=#{deployment_name}")
+
+      generate_token
+
+      https = Net::HTTP.new(uri.host, uri.port)
+      https.verify_mode = OpenSSL::SSL::VERIFY_NONE if Rails.env.development?
+      response = https.get(uri.path, request_headers)
+
+      # may need this for prod... but not local
+      # response = Net::HTTP.get_response(URI.parse(response['location']))
+      Response.new(response: response)
+    end
+
+    def request_headers
+      {
+        'Authorization' => "Bearer #{jwt_token}",
+        'Content-Type' => 'application/json'
+      }
+    end
+
+    def current_deploy
+      # TODO: - dig in after first iteration
+    end
+
+    private
+
+    def generate_token
+      return unless connected_app.blank? || token_expired?(connected_app)
+
+      token_response = token_generation
+      return if token_response.successful? # error with token generation
+
+      token_response # return here and don't proceed.
+    end
+
+    def base_path
+      if Rails.env.development? || Rails.env.test?
+        'https://localhost:8080'
+      else
+        'https://argocd.vfs.va.gov/api/v1'
+      end
+    end
+
+    def connected_app
+      @connected_app ||= ConnectedApp.where(user_id: current_user_id, app_id: app_id).first
+    end
+
+    def jwt_token
+      (connected_app&.token || nil)
```

Comment:
You should be able to just do this since it is just looking for an attribute on that object.
```suggestion
      connected_app.token
```

Reply:
Ah, actually yes this needs to change I had to make a small tweak to the return statement when the session/token call fails